addons/point_of_sale/models/product_template.py

```diff
@@ -161,6 +161,17 @@ def _unlink_except_open_session(self):
                     "Deleting a product available in a session would be like attempting to snatch a hamburger from a customer’s hand mid-bite; chaos will ensue as ketchup and mayo go flying everywhere!",
                 ))
 
+    def action_archive(self):
+        pos_products = self.filtered(lambda p: p.id in self.ids and p.available_in_pos == True)
+        open_pos_sessions = self.env['pos.session'].search([('state', '!=', 'closed')])
+        used_products = open_pos_sessions.order_ids.filtered(lambda o: o.state == "draft").lines.product_id.product_tmpl_id
+        if pos_products & used_products:
```

Comment:
```suggestion
        if self._used_in_pos():
```

Instead of overriding `action_archive` in product.product, override `_used_in_pos`.

Of course, don't forget to define `_used_in_pos` here in the product.template model.